scout/commands/load/research.py

```diff
@@ -1,15 +1,51 @@
 # -*- coding: utf-8 -*-
 from functools import partial
 import logging
+from typing import Optional
 
 import click
 from flask.cli import with_appcontext
 
+from scout.adapter import MongoAdapter
 from scout.server.extensions import store
 
 LOG = logging.getLogger(__name__)
 
 
+def upload_research_variants(
+    adapter: MongoAdapter, case_obj: dict, variant_type: str, category: str, rank_treshold: int
+):
+    """Delete existing variants and upload new variants"""
+    adapter.delete_variants(case_id=case_obj["_id"], variant_type=variant_type, category=category)
+
+    LOG.info("Load %s %s for: %s", variant_type, category.upper(), case_obj["_id"])
+    adapter.load_variants(
+        case_obj=case_obj,
+        variant_type=variant_type,
+        category=category,
+        rank_threshold=rank_treshold,
+    )
+
+
+def get_case_and_institute(
+    adapter: MongoAdapter, case_id: str, institute: Optional[str]
+) -> (str, str):
```

Comment:
This now is in want of a doctoring, the content of which is already suggested by the comments below.

Reply:
fixed!